beaconChain/tests/utils/state.ts

```diff
@@ -0,0 +1,206 @@
+import BN from "bn.js";
+
+import {SHARD_COUNT, GENESIS_EPOCH, ZERO_HASH, GENESIS_SLOT, GENESIS_FORK_VERSION, GENESIS_START_SHARD, LATEST_RANDAO_MIXES_LENGTH, LATEST_BLOCK_ROOTS_LENGTH, LATEST_INDEX_ROOTS_LENGTH, LATEST_PENALIZED_EXIT_LENGTH} from "../../constants";
+import {BeaconState, Crosslink, Eth1Data, Fork, PendingAttestation, uint64, Validator} from "../../types";
+import {Eth1DataVote, bytes32} from "../../types";
+import {generateValidators} from "./validator";
+import {randBetween, randBetweenBN} from "./misc";
+
+/**
+ * Copy of BeaconState, but all fields are marked optional to allow for swapping out variables as needed.
+ */
+interface TestBeaconState {
+  // Misc
+  slot?: uint64;
+  genesisTime?: uint64;
+  fork?: Fork; // For versioning hard forks
+
+  // Validator registry
+  validatorRegistry?: Validator[];
+  validatorBalances?: uint64[];
+  validatorRegistryUpdateEpoch?: uint64;
+
+  // Randomness and committees
+  latestRandaoMixes?: bytes32[];
+  previousEpochStartShard?: uint64;
+  currentEpochStartShard?: uint64;
+  previousCalculationEpoch?: uint64;
+  currentCalculationEpoch?: uint64;
+  previousEpochSeed?: bytes32;
+  currentEpochSeed?: bytes32;
+
+  // Finality
+  previousJustifiedEpoch?: uint64;
+  justifiedEpoch?: uint64;
+  justificationBitfield?: uint64;
+  finalizedEpoch?: uint64;
+
+  // Recent state
+  latestCrosslinks?: Crosslink[];
+  latestBlockRoots?: bytes32[];
+  latestIndexRoots?: bytes32[];
+  latestPenalizedBalances?: uint64[]; // Balances penalized at every withdrawal period
+  latestAttestations?: PendingAttestation[];
+  batchedBlockRoots?: bytes32[];
+
+  // Ethereum 1.0 deposit root
+  latestEth1Data?: Eth1Data;
+  eth1DataVotes?: Eth1DataVote[];
+}
+
+/**
+ * Generate beaconState, by default it will use the initial state defined when the `ChainStart` log is emitted.
+ * NOTE: All fields can be overridden through `opts`.
+ * @param {TestBeaconState} opts
+ * @returns {BeaconState}
+ */
+export function generateState(opts?: TestBeaconState): TestBeaconState {
```

Comment:
I think this should return a `BeaconState` like generateRandomState?

Consider renaming to generateZeroedState or in comments describe how this function and generateRandomState mirror each other/ are different.